mesonbuild/backend/backends.py

```diff
@@ -1829,8 +1823,6 @@ def generate_header_install(self, d: InstallData) -> None:
                     outdir_name = os.path.join('{includedir}', subdir)
 
             for f in h.get_sources():
-                if not isinstance(f, File):
-                    raise MesonException(f'Invalid header type {f!r} can\'t be installed')
```

Comment:
Note to self: this dead code was made impossible in 251dff56fb074cd718da81c21e104cd26d897869

Reply:
Incorportated into commit message.